analyzers/tests/SonarAnalyzer.UnitTest/PackagingTests/RuleTypeTest.cs

```diff
@@ -52,15 +48,18 @@ private static void DetectTypeChanges(Dictionary<string, RuleDescriptor> rules,
                                   .Where(x => x.ActualType != x.ExpectedType)
                                   .ToList();
 
-            var newRules = items.Where(x => x.ExpectedType is null);
-            newRules.Should().BeEmpty($"you need to add the rules in {expectedTypesName}");
+            var newRules = rulesWithUnmatchingType.Where(x => x.ExpectedType is null);
+            newRules
+                .Should()
+                .BeEmpty($": there are rules that exist in '{language}::SonarAnalyzer.RuleCatalog.Rules' file but not in {expectedTypesName}. You need to add them to the latter");
 
-            items.Should().NotContain(
-                x => x.ActualType == null && !DeletedRules.Contains($"S{x.RuleId}"),
-                "YOU SHOULD NEVER DELETE RULES! Rules should not be deleted without careful consideration and prior deprecation. We need to notify the product teams (SQ, SC, SL) as well.");
+            var ruleWithoutActualTypeExists = rulesWithUnmatchingType.Exists(x => x.ActualType is null);
+            ruleWithoutActualTypeExists
+                .Should()
+                .BeFalse($": there are rules that exist in {expectedTypesName} but not in '{language}::SonarAnalyzer.RuleCatalog.Rules' file. You might have forgotten to update the RSpec for sonar-dotnet");
```

Comment:
This could happen when rule ID is removed. That should be rare, so I wouldn't give clues. RSpec update would not fix this.
```suggestion
                .BeFalse($": there are rules that exist in {expectedTypesName} but not in '{language}::SonarAnalyzer.RuleCatalog.Rules' file.");
```